test/sharness/t0250-files-api.sh

```diff
@@ -0,0 +1,219 @@
+#!/bin/sh
+#
+# Copyright (c) 2015 Jeromy Johnson
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="test the unix files api"
+
+. lib/test-lib.sh
+
+test_init_ipfs
+
+# setup files for testing
+test_expect_success "can create some files for testing" '
+	FILE1=$(echo foo | ipfs add -q) &&
+	FILE2=$(echo bar | ipfs add -q) &&
+	FILE3=$(echo baz | ipfs add -q) &&
+	mkdir stuff_test &&
+	echo cats > stuff_test/a &&
+	echo dogs > stuff_test/b &&
+	echo giraffes > stuff_test/c &&
+	DIR1=$(ipfs add -q stuff_test | tail -n1)
+'
+
+verify_path_exists() {
+	# simply running ls on a file should be a good 'check'
+	ipfs files ls $1
+}
+
+verify_dir_contents() {
+	dir=$1
+	shift
+	rm -f expected
+	touch expected
+	for e in $@
+	do
+		echo $e >> expected
+	done
+
+	test_expect_success "can list dir" '
+		ipfs files ls $dir > output
+	'
+
+	test_expect_success "dir entries look good" '
+		test_sort_cmp output expected
+	'
+}
+
+test_files_api() {
+	test_expect_success "can mkdir in root" '
+		ipfs files mkdir /cats
+	'
+
+	test_expect_success "directory was created" '
+		verify_path_exists /cats
+	'
+
+	test_expect_success "directory is empty" '
+		verify_dir_contents /cats
+	'
+
+	test_expect_success "can put files into directory" '
+		ipfs files cp /ipfs/$FILE1 /cats/file1
+	'
+
+	test_expect_success "file shows up in directory" '
+		verify_dir_contents /cats file1
+	'
+
+	test_expect_success "can read file" '
+		ipfs files read /cats/file1 > file1out
+	'
+
+	test_expect_success "output looks good" '
+		echo foo > expected &&
+		test_cmp file1out expected
```

Comment:
i think the convention is `test_cmp expected actual`. it's already helped catch bugs